### PR TITLE
Qurt lightweight parameter implementation

### PR DESCRIPTION
**Pull request by ZacharyLowell**
### Solved Problem
Originally the parameter PR for getting qurt to compile and sync was quite large. This is a much lighter weight version that just allows params to compile and run on the dsp/apps architecture but there is currently no syncing of parameters between the two sides.

### Solution
Added in the necessary dependencies to allow for params to compile on qurt.

### Alternatives
Eventually we need to setup the client server model to sync the parameters between apps and dsp.

### Test coverage
Tested on voxl2.

### CODE REVIEW

**Review by katzfey — DISMISSED** [review]

File: platforms/common/include/px4_platform_common/posix.h [hunk 1]
```diff
@@ -114,6 +114,11 @@ __END_DECLS
 #error "No TARGET OS Provided"
 #endif
 
+#if defined(__PX4_QURT)
```
Comment (dagar) [inline platforms/common/include/px4_platform_common/posix.h:117]: Can you merge this into the ifdefs above where we already check if PX4_NUTTX, PX4_POSIX, etc (and error if no target OS provided).

Comment (ZacharyLowell) [reply platforms/common/include/px4_platform_common/posix.h:117]: Yep! Will do that asap.

**Comment by ZacharyLowell** [comment]
@dagar I moved the fsync code into the comparisons above. Only problem is that qurt leverages both posix and qurt architecture during the build so both PX4_POSIX and PX4_QURT are true so I had to put the PX4_QURT evaluator inside the PX4_POSIX code so it could be read aswell. Not another elif.

**Review by katzfey — APPROVED** [review]

**Comment by dagar** [comment]
Alright, thanks for trying.

Ultimately I think we should do a complete split where each platform explicitly provides these headers. In the cases where things are common/shared we could just have the platform header include the relevant piece.

**Pull request merged**